bin/internal/update_dart_sdk.ps1

```diff
@@ -33,14 +34,16 @@ if ((Test-Path "$flutterRoot\DEPS" -PathType Leaf) -and (Test-Path "$flutterRoot
 
     if (($branch -ne "stable" -and $branch -ne "beta")) {
         # Write the engine version out so downstream tools know what to look for.
-        $engineVersion | Out-File -FilePath "$flutterRoot\bin\internal\engine.version"
+        [System.IO.File]::WriteAllText("$flutterRoot\bin\internal\engine.version", $engineVersion, [System.Text.Encoding]::UTF8)
```

Comment:
This instance of the UTF8 encoding is prepending a byte order mark to the file.

The `engine.version` in the current framework repository does not have that.

I don't know of a specific problem that would be caused by the BOM.  But I'd suggest removing it in case some consumer of this file depends on the old format.

Reply:
That will probably be a problem. I remember hitting this a while ago - the documentation between PowerShell 5.1 and 7+ and what gets written is not the greatest.

Reply:
Going to try with $utf8NoBom = New-Object System.Text.UTF8Encoding($false)